estate/models/estate_property.py

```diff
@@ -0,0 +1,128 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _
+from odoo import api
+from odoo import fields
+from odoo import models
+from odoo.exceptions import UserError
+from odoo.exceptions import ValidationError
+from odoo.tools import float_is_zero
+
+class EstateProperty(models.Model):
+    _name = 'estate.property'
+    _description = "Estate Properties"
+    _order = 'id desc'
+    _sql_constraints = [
+        ('check_expected_price', "CHECK(expected_price > 0)",
+        "The expected price must be strictly positive"),
+        ('check_selling_price', "CHECK(selling_price >= 0)",
+        "The selling price must be positive"),
+    ]
+
+    name = fields.Char(string="Title", required=True)
+    description = fields.Text()
+    image = fields.Binary(string="Property Image")
+    postcode = fields.Char()
+    date_availability = fields.Date(
+        string="Available From",
+        default=lambda self:fields.Date.add(fields.Date.today(), months=3),
+        copy=False
+        )
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(index=True, default=2)
+    living_area = fields.Integer(string="Living Area (sqm)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer(string="Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        selection=[
+            ('north', 'North'),
+            ('south', 'South'),
+            ('east', 'East'),
+            ('west', 'West'),
+        ])
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('offer received', 'Offer Received'),
+            ('offer accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        default='new',
+        copy=False
+    )
+    property_type_id = fields.Many2one(comodel_name='estate.property.type', string="Property Type")
+    buyer_id = fields.Many2one(comodel_name='res.partner', string="Buyer", copy=False)
+    salesman_id = fields.Many2one(comodel_name='res.users', string="Salesman", default=lambda self:self.env.uid)
+    tag_ids = fields.Many2many('estate.property.tag')
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string="offer_id")
+    total_area = fields.Integer(compute='_compute_total_area', string="Total Area (sqm)")
+    best_offer = fields.Float(compute='_compute_best_offer', string="Best Offer", store=True)
+    company_id = fields.Many2one(comodel_name='res.company',required=True, default=lambda self:self.env.company)
```

Comment:
```suggestion
    company_id = fields.Many2one(comodel_name='res.company', required=True, default=lambda self:self.env.company)
```
Make sure to add a space after comma and no space before comma.